### PR TITLE
[ANCHOR-173] Removed data.url and add data.server and data.database fields

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

This is to replace `data.url` with `data.server` and `data.database` fields.

Here is the original data configuration:
```
  data: 
    url: jdbc:postgresql://127.0.0.1:5432/postgres
```

Here is the data configuration after the refactoring.

```
data:
  type: postgres
  server: 127.0.0.1:5432
  database: postgres
```

### Why

This is to avoid mistakes while constructing database url string to reduce the user friction. 

### Known limitations

N/A

### CODE REVIEW

File: platform/src/main/resources/config/anchor-config-default-values.yaml [hunk 1]
```diff
@@ -485,11 +485,19 @@ data:
...
   ## @type: string
   ## Location of the database
   #
-  url: jdbc:h2:mem:anchor-platform
+#  url: jdbc:h2:mem:anchor-platform
+
+  ## The hostname of the database server.
+  server:
+
+  ## Name of the database. Optional. Default to anchor-platform or the default name of the type of database.
```
Comment (JakeUrban) [inline platform/src/main/resources/config/anchor-config-default-values.yaml:498]: Does this mean in postgres, it defaults to "postgres"? What do you think about requiring this field instead?

File: platform/src/main/resources/config/anchor-config-default-values.yaml [hunk 2]
```diff
@@ -485,11 +485,19 @@ data:
...
+
   ## @param: url
   ## @type: string
   ## Location of the database
   #
-  url: jdbc:h2:mem:anchor-platform
+#  url: jdbc:h2:mem:anchor-platform
+
+  ## The hostname of the database server.
+  server:
```
Comment (JakeUrban) [inline platform/src/main/resources/config/anchor-config-default-values.yaml:496]: what would the value be for databases that don't use a server, like h2 or sqlite?

**Review by JakeUrban — COMMENTED** [review]
Left some questions, but more generally, aren't we missing changes to other files to process these values?

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 2 above)
Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:496]: If type is h2, the server and database will be ignore.

sqlite has a database name.

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 1 above)
Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:498]: I think we should make it required. If the database does not exist, AP will fail. 

We can add automatic database creation if that is becoming an issue.

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 2 above)
Comment (JakeUrban) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:496]: Sounds straightforward!

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 1 above)
Comment (JakeUrban) [reply platform/src/main/resources/config/anchor-config-default-values.yaml]: Awesome, lets do it.

**Review by Ifropc — APPROVED** [review]

**Review by JakeUrban — APPROVED** [review]

File: platform/src/main/resources/config/anchor-config-default-values.yaml [hunk 3]
```diff
@@ -485,11 +485,13 @@ data:
...
-  ## @type: string
-  ## Location of the database
-  #
-  url: jdbc:h2:mem:anchor-platform
+
+  ## The hostname and port of the database server.
+  server:
+
+  ## Name of the database.
+  database:
```
Comment (JakeUrban) [inline platform/src/main/resources/config/anchor-config-default-values.yaml:493]: What do you think about using `dbName` here?

Comment (JakeUrban) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:493]: Not a big deal, just an idea. Its not totally clear that you're looking for the database name when I use the `DATA_DATABASE` environment variable.

Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:493]: from Postgres doc: https://jdbc.postgresql.org/documentation/use/

They use `database` as the database name. I think we can keep database unless you feel strong about this.

Comment (JakeUrban) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:493]: Nope, thats fine.

**Pull request merged**